.github/workflows/publish.yml

```diff
@@ -0,0 +1,53 @@
+name: Publish Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      version_bump:
+        description: 'Version bump (major, minor, patch)'
+        required: true
+        default: 'patch'
+
+jobs:
+  validate-input:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Validate input
+        id: validate
+        run: |
+          if [[ "${{ github.event.inputs.version_bump }}" != "major" && "${{ github.event.inputs.version_bump }}" != "minor" && "${{ github.event.inputs.version_bump }}" != "patch" ]]; then
+            echo "Invalid input for version_bump. Please provide 'major', 'minor', or 'patch'."
+            exit 1
+          fi
+
+  release:
+    needs: validate-input
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - id: release
+        uses: rymndhng/release-on-push-action@v0.28.0
+        with:
+          bump_version_scheme: ${{ github.event.inputs.version_bump }}
+          use_github_release_notes: true
+          tag_prefix: v
```

Comment:
i think this repo uses this static file for version bump https://github.com/teslamotors/vehicle-command/blob/main/pkg/account/version.txt#L1
not sure if this will work as intended

Reply:
Yeah. I'm open to suggestions on how to manage this better, but the idea is to have client User Agent versions correspond to SDK versions. The Makefile updates it but `go build`/`go get` are more common than `make`.

Reply:
Revised `publish.yml` so it automatically creates a GitHub release and pushes to Docker hub when the `version.txt` file changes. 

Open to suggestions on how to test the workflow before merging.

Reply:
Just tested this in a private repository and everything is working well.

GitHub release is auto created and the image is pushed to Docker with `latest` tag and version tag. When something is merged and version tag doesn't change, nothing is triggered.

Reply:
We will need to make sure the Docker secrets are set and the workflow runner has write access to the repository.